kajas/src/app/setup-profile/setup-profile.component.ts

```diff
@@ -15,64 +15,76 @@ export class SetupProfileComponent implements OnInit {
   countries: any[] = [];
   cities: any[] = [];
   selectedCountryName = '';
+  showModal = false;
+  modalMessage = '';
 
-  constructor(private fb: FormBuilder, private locationService: LocationService, 
-    private sessionStorage: SessionStorageService, private router: Router) {
+  firstNamePlaceholder = this.sessionStorage.get('first_name') || '';
+  lastNamePlaceholder = this.sessionStorage.get('last_name') || '';
+  middleNamePlaceholder = this.sessionStorage.get('middle_name') || '';
+  emailPlaceholder = this.sessionStorage.get('email') || '';
+  kajasPlaceholder = this.sessionStorage.get('username') || '';
 
+  constructor(
+    private fb: FormBuilder, 
+    private locationService: LocationService, 
+    private sessionStorage: SessionStorageService, 
+    private router: Router
+  ) {
     this.profileForm = this.fb.group({
-      id: this.sessionStorage.get('id'),
+      id: [this.sessionStorage.get('id')],
       profile: [''],
-      bio: [''],
-      firstName: ['', {
-        validators: [
-          Validators.required
-        ]
-      }],
-      lastName: ['', {
-        validators: [
-          Validators.required
-        ]
-      }],
-      middleName: [''],
-      email: ['', {
-        validators: [
-          Validators.required,
-          Validators.email
-        ]
-      }],
-      country: ['', {
-        validators: [
-          Validators.required
-        ]
-      }],
-      city: ['', {
-        validators: [
-          Validators.required
-        ]
-      }],
-      linkedIn: [''],
-      facebook: [''],
-      instagram: [''],
-      website: [''],
-      kajas_link: this.sessionStorage.get('username')
+      bio: ['', [Validators.maxLength(250)]],
+      firstName: [this.sessionStorage.get('first_name'), [Validators.required]],
+      lastName: [this.sessionStorage.get('last_name'), [Validators.required]],
+      middleName: [this.sessionStorage.get('middle_name')],
+      email: [{ value: this.sessionStorage.get('email'), disabled: true }],
+      country: ['', [Validators.required]],
+      city: ['', [Validators.required]],
+      linkedIn: ['', [this.urlValidator(), this.socialMediaValidator('linkedin')]],
```

Comment:
### Always use the object form in creating validators.
https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g270ee9174f4_0_43